test/parallel/test-assert.js

```diff
@@ -1015,10 +1023,43 @@ common.expectsError(
   {
     code: 'ERR_ASSERTION',
     type: assert.AssertionError,
-    message: 'test'
+    message: 'test',
+    generatedMessage: false
   }
 );
 
+// Do not try to check Node.js modules.
+{
+  const e = new EventEmitter();
+
+  e.on('hello', assert);
+
+  let threw = false;
+  try {
+    e.emit('hello', false);
+  } catch (err) {
+    const frames = err.stack.split('\n');
+    const [, filename, line, column] = frames[1].match(/\((.+):(\d+):(\d+)\)/);
+    // Reset the cache to check again
+    errorCache.delete(`${filename}${line - 1}${column - 1}`);
+    const data = `${'\n'.repeat(line - 1)}${' '.repeat(column - 1)}` +
+                 'ok(failed(badly));';
+    try {
+      writeFileSync(filename, data);
```

Comment:
Oh, I just saw this right now. What this test makes sure is that if `assert.ok` is triggered inside of Node.js core code and that Node.js core file has the same name as a user file in the root directory (in this example `events.js`) it would have read the users file and that is clearly wrong.